src/libYARP_dev/src/devices/RobotDescriptionServer/RobotDescriptionServer.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2016 iCub Facility - Istituto Italiano di Tecnologia
+ * Author: Marco Randazzo <marco.randazzo@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <yarp/dev/IRobotDescription.h>
+#include "RobotDescriptionServer.h"
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/LockGuard.h>
+
+/*! \file Navigation2DClient.cpp */
+
+using namespace yarp::dev;
+using namespace yarp::os;
+using namespace yarp::sig;
+
+
+//------------------------------------------------------------------------------------------------------------------------------
+
+bool yarp::dev::RobotDescriptionServer::open(yarp::os::Searchable &config)
+{
+    m_local_name.clear();
+    m_local_name = config.find("local").asString().c_str();
+
+    if (m_local_name == "")
+    {
+        yError("RobotDescriptionServer::open() error you have to provide valid local name");
+        return false;
+    }
+
+    ConstString local_rpc = m_local_name;
+    local_rpc += "/rpc";
+
+    if (!m_rpc_port.open(local_rpc.c_str()))
+    {
+        yError("RobotDescriptionServer::open() error could not open rpc port %s, check network", local_rpc.c_str());
+        return false;
+    }
+
+    m_rpc_port.setReader(*this);
+    return true;
+}
+
+bool yarp::dev::RobotDescriptionServer::attachAll(const PolyDriverList &p)
+{
+    for (int i = 0; i < p.size(); i++)
+    {
+        //yDebug() << "***************" << p[i]->poly->getOptions().toString();
+        //yDebug() << "***************" << p[i]->poly->getValue("device").toString();
+        //yDebug() << "***************" << p[i]->poly->getValue("name").toString();
+        RobotDescription dev;
+        dev.device_name = p[i]->poly->getValue("name").toString();
```

Comment:
Ooohh, I didn't know about this feature... kind of cool.

Anyway we could either:
- Choose a keyword for this type of interaction, like `deviceDescription` or similar (to be unique and not used by any device right now)
- Have the devices implement the `robotDescription` interface  and then during the attach we can check if the device has the interface call the `getAllDevices` on it.

I prefer the second one, because:
- it is more clear
-  does not force a specific keyword in the config file,
- it can be implemented by all devices, not only the wrappes.

If a device does not have the interface yet, it'll be skipped or fallback trying to guess the information from parameters as it is doing now.